cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/resource/ConfigMapOperator.java

```diff
@@ -29,4 +30,18 @@ public ConfigMapOperator(Vertx vertx, KubernetesClient client) {
     protected MixedOperation<ConfigMap, ConfigMapList, DoneableConfigMap, Resource<ConfigMap, DoneableConfigMap>> operation() {
         return client.configMaps();
     }
+
+    @Override
+    protected Future<ReconcileResult<ConfigMap>> internalPatch(String namespace, String name, ConfigMap current, ConfigMap desired) {
+        try {
+            if (current.getData().equals(desired.getData())) {
+                log.debug("{} {} in namespace {} has not been patched because resources are equal", resourceKind, name, namespace);
+                return Future.succeededFuture(ReconcileResult.noop());
```

Comment:
This still won't patch when the `data` are equal. That's wrong. Consider what happens if the user adds a label to their `Kafka` resource. What should happen is all the dependent resources get the same label. If we don't call `internalPatch()` the CM won't have its label changed. So you should always call `super.internalPatch()`, but use the `data` maps to determine whether to _say_ the result was a no op. It's a lie, but I think it's acceptable (at least at the moment).

Reply:
This still needs fixing.